src/core/vectortile/qgsvectortileconnection.h

```diff
@@ -0,0 +1,55 @@
+/***************************************************************************
+    qgsvectortileconnection.h
+    ---------------------
+    begin                : March 2020
+    copyright            : (C) 2020 by Martin Dobias
+    email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#ifndef QGSVECTORTILECONNECTION_H
+#define QGSVECTORTILECONNECTION_H
+
+#include "qgis_core.h"
+
+///@cond PRIVATE
+#define SIP_NO_FILE
+
+#include <QStringList>
+
+struct QgsVectorTileConnection
+{
+  QString name;
+  QString url;
+  int zMin = -1;
+  int zMax = -1;
+
+  QString encodedUri() const;
+};
+
+//! Utility class for handling list of connections to vector tile layers
+class CORE_EXPORT QgsVectorTileConnectionUtils
```

Comment:
Hmm I have thought that class was meant mainly for databases - because for example raster XYZ tile connections were not transformed to use QgsAbstractProviderConnection - and I used code from there for vector tile connections. Any thoughts @elpaso ?

Reply:
The class is designed for everything, it's just that only a few providers have implemented it so far

Reply:
Ah only now I see there are some fresh additions like QgsProviderConnectionModel and QgsProviderConnectionComboBox - things are starting to make sense - thanks for the pointer :-)

Reply:
Yep, it's the benefit of moving to a common API. Still early days, but we'll see much more development on this front in upcoming releases...